x/evm/keeper/migrate.go

```diff
@@ -2,69 +2,99 @@ package keeper
 
 import (
 	"encoding/hex"
-	"fmt"
-	"strings"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
 	"github.com/axelarnetwork/axelar-core/utils"
+	"github.com/axelarnetwork/axelar-core/utils/key"
 	"github.com/axelarnetwork/axelar-core/x/evm/types"
-	"github.com/axelarnetwork/axelar-core/x/nexus/exported"
+	"github.com/axelarnetwork/utils/funcs"
 	"github.com/axelarnetwork/utils/slices"
 )
 
-// Migrate7To8 returns the handler that performs in-place store migrations
-func Migrate7To8(k *BaseKeeper, n types.Nexus) func(ctx sdk.Context) error {
+// Migrate8to9 returns the handler that performs in-place store migrations
+func Migrate8to9(bk *BaseKeeper, n types.Nexus) func(ctx sdk.Context) error {
 	return func(ctx sdk.Context) error {
-		chains := slices.Filter(n.GetChains(ctx), func(chain exported.Chain) bool { return chain.Module == types.ModuleName })
-		for _, chain := range chains {
-			if err := migrateBurnerInfoForChain(ctx, k, chain, burnerAddrPrefixDeprecated); err != nil {
+		for _, chain := range slices.Filter(n.GetChains(ctx), types.IsEVMChain) {
+			ck := funcs.Must(bk.ForChain(ctx, chain.Name)).(chainKeeper)
+
+			if err := migrateDeposits(ctx, ck, types.DepositStatus_Confirmed); err != nil {
 				return err
 			}
-			if err := migrateBurnerInfoForChain(ctx, k, chain, strings.ToLower(burnerAddrPrefixDeprecated)); err != nil {
+
+			if err := migrateDeposits(ctx, ck, types.DepositStatus_Burned); err != nil {
 				return err
 			}
-
-			k.Logger(ctx).Info(fmt.Sprintf("migrated all burner info keys for chain %s", chain.Name))
-
 		}
-		k.Logger(ctx).Info("burner info keys migration complete")
+
 		return nil
 	}
 }
 
-func migrateBurnerInfoForChain(ctx sdk.Context, k *BaseKeeper, chain exported.Chain, oldKey string) error {
-	ck, err := k.forChain(ctx, chain.Name)
-	if err != nil {
-		return err
+func migrateDeposits(ctx sdk.Context, ck chainKeeper, status types.DepositStatus) error {
+	var prefix key.Key
+	switch status {
+	case types.DepositStatus_Confirmed:
+		prefix = key.FromStr(confirmedDepositPrefixDeprecated)
+	case types.DepositStatus_Burned:
+		prefix = key.FromStr(burnedDepositPrefixDeprecated)
 	}
 
-	// migrate in batches so memory pressure doesn't become too large
-	keysToDelete := make([][]byte, 0, 1000)
-	for {
-		iterBurnerAddr := ck.getStore(ctx).Iterator(utils.KeyFromStr(oldKey))
+	iter := ck.getStore(ctx).IteratorNew(prefix)
+	defer utils.CloseLogError(iter, ck.Logger(ctx))
 
-		if !iterBurnerAddr.Valid() {
-			return iterBurnerAddr.Close()
+	for ; iter.Valid(); iter.Next() {
+		var deposit types.ERC20Deposit
+		iter.UnmarshalValue(&deposit)
+
+		transferEvents := getTransferEventsByTxIDAndAddress(ctx, ck, deposit.TxID, deposit.BurnerAddress)
+		if len(transferEvents) == 0 {
+			// Deposits from the time when we have not started doing event processing.
+			// Their log indexes are no way to be retrieved anymore and therefore ignore.
+			continue
```

Comment:
count such occurrences, and log them? I'm particularly interested to know if any of the deposits in confirmed status satisfied this, since they won't get handled then.